src/test/java/se/sundsvall/snailmail/api/SnailMailResourceTest.java

```diff
@@ -1,39 +1,100 @@
 package se.sundsvall.snailmail.api;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.doNothing;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static se.sundsvall.snailmail.TestDataFactory.buildSendSnailMailRequest;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.zalando.problem.Status.BAD_REQUEST;
+
+import java.util.UUID;
 
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.http.HttpStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.zalando.problem.violations.ConstraintViolationProblem;
+import org.zalando.problem.violations.Violation;
 
+import se.sundsvall.snailmail.Application;
 import se.sundsvall.snailmail.api.model.SendSnailMailRequest;
 import se.sundsvall.snailmail.service.SnailMailService;
 
-@ExtendWith(MockitoExtension.class)
+@SpringBootTest(classes = Application.class, webEnvironment = RANDOM_PORT)
+@ActiveProfiles("junit")
 class SnailMailResourceTest {
 
-	@Mock
+	@MockBean
 	private SnailMailService mockSnailMailService;
 
-	@InjectMocks
-	private SnailMailResource snailMailResource;
+
+	@Autowired
+	private WebTestClient webTestClient;
+
 
 	@Test
 	void sendSnailMail() {
-		doNothing().when(mockSnailMailService).sendSnailMail(any(SendSnailMailRequest.class));
 
-		final var response = snailMailResource.sendSnailMail(buildSendSnailMailRequest());
-		assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+		//ACT
+		webTestClient.post()
+			.uri("/send/snailmail")
+			.bodyValue(SendSnailMailRequest.builder()
+				.withDepartment("department")
+				.build())
+			.exchange()
+			.expectStatus()
+			.isOk();
 
+		// VERIFY
 		verify(mockSnailMailService, times(1)).sendSnailMail(any(SendSnailMailRequest.class));
```

Comment:
`times(1)` is redundant and can be removed (goes for all places where it is used)